luigi/contrib/redshift.py

```diff
@@ -308,12 +324,18 @@ def create_table(self, connection):
                     type=type,
                     encoding=encoding) for name, type, encoding in self.columns
             )
+
+            table_constraints = ''
+            if self.table_constraints != '':
+                table_constraints = ',' + self.table_constraints
+
             query = ("CREATE {type} TABLE "
-                     "{table} ({coldefs}) "
+                     "{table} ({coldefs} {table_constraints}) "
                      "{table_attributes}").format(
                 type=self.table_type,
                 table=self.table,
                 coldefs=coldefs,
+                table_constraints=table_constraints,
```

Comment:
Out of the scope of this PR, but if you'd like it would be awesome to reduce the duplicated code here for where len == 2 and len == 3 for the creation of `query`. The only difference is the definition of `coldefs`

Reply:
I would agree with you that the whole Redshift contrib should be refactored. There would a lot of value to make it a little bit more DRY.

If I have more time in the upcoming weeks I'll take a jab at refactoring it so that it's a little bit saner! :)